site/content/integrate/apps/quick-start-serverless/_index.md

```diff
@@ -0,0 +1,443 @@
+	InitApp(apps.DeployOpenFAAS)
+	http.DefaultServeMux.ServeHTTP(w, r)
+}
+```
+
+Note: names `package function` and `func Handle` appear to be required
+(hardcoded) for OpenFaaS.
+
+`golang-middleware` template makes use of adding boilerplate handlers to
+`http.DefaultServeMux`, so we just need to add our handlers and serve.
+
+```go
+var deployType apps.DeployType
+
+func InitApp(dt apps.DeployType) {
+	// Returns the Channel Header and Command bindings for the App.
+	http.HandleFunc("/bindings", httputils.HandleJSONData(bindingsData))
+
+	// The form for sending a Hello message.
+	http.HandleFunc("/send/form", httputils.HandleJSONData(formData))
+
+	// The main handler for sending a Hello message.
+	http.HandleFunc("/send/submit", send)
+
+	deployType = dt
+}
+
+func send(w http.ResponseWriter, req *http.Request) {
+	creq := apps.CallRequest{}
+	json.NewDecoder(req.Body).Decode(&creq)
+
+	message := fmt.Sprintf("Hello from a serververless app running as %s!", deployType)
+	v, ok := creq.Values["message"]
+	if ok && v != nil {
+		message += fmt.Sprintf(" ...and %s!", v)
+	}
+
+	asBot := appclient.AsBot(creq.Context)
+	asBot.DM(creq.Context.ActingUserID, message)
+
+	httputils.WriteJSON(w,
+		apps.NewOKResponse(nil, "Created a post in your DM channel."))
+}
+```
+
+#### `go.mod` and `go.sum`
+
+Since package `function` is built independently by OpenFaaS, add its own `go.mod` for its ddependencies:
+
+```go
+module {{path-to-your-module}}/function
+
+go 1.16
+
+require github.com/mattermost/mattermost-plugin-apps v0.8.0
+```
+
+### 4. `./aws`
+#### `main.go`
+```go
+package main
+
+import (
+    "net/http"
+
+    "github.com/aws/aws-lambda-go/lambda"
+    "github.com/awslabs/aws-lambda-go-api-proxy/httpadapter"
+    "github.com/mattermost/mattermost-plugin-apps/apps"
+
+    "{{path-to-your-module}}/function"
+)
+
+func main() {
+    function.InitApp(apps.DeployAWSLambda)
+
+    // run with the Lambda request re-marshaled back into an http.Request so that it works with mux.
+    lambda.Start(httpadapter.New(http.DefaultServeMux).Proxy)
+}
+```
+
+#### `go.mod`
+```go
+module {{path-to-your-module}}/aws
+
+go 1.16
+
+require (
+	github.com/aws/aws-lambda-go v1.19.1
+	github.com/awslabs/aws-lambda-go-api-proxy v0.11.0
+	github.com/mattermost/mattermost-plugin-apps v0.8.0
+	{{path-to-your-module}}/function v0.0.0-00010101000000-000000000000
+)
+
+replace {{path-to-your-module}}/function => ../function
+```
+
+### 5. `./http`
+#### `main.go`
+The HTTP main serves `./static` and `manifest.json` in addition to the app call paths.
+
+```go
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
+
+	"{{path-to-your-module}}/function"
+)
+
+func main() {
+	mpath := flag.String("manifest", "", "path to the manifest file to serve on /manifest.json")
+	spath := flag.String("static", "", "path to the static folder to serve on /static")
```

Comment:
What do these configuration parameters add for the user? Are they worth the additional complexity?

Reply:
No, but I wanted to keep the `manifest.json` and `/static` at the top of the app folder/bundle; this was my workaround `embed` not letting me go `../`

I could just default these to `../xyz`? Hardcode?

Reply:
0/5 the root directly could contain a `go` file that embeds `manfest.json` and `static`. `http` then imports them.